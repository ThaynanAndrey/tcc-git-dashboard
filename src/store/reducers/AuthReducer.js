import { USER_LOGIN_SUCCESS, USER_LOGIN_ERROR, USER_LOGOUT_SUCCESS, USER_LOGOUT_ERROR } from '../actions/types';

export const initState = {
    user: undefined,
    accessToken: undefined,
    errorLogin: undefined,
    errorLogout: undefined
};
  
const authReducer = (state = initState, action) => {
    switch(action.type) {
        case USER_LOGIN_SUCCESS:
            return {
                ...state,
                user: action.user,
                idUser: action.idUser,
                accessToken: action.accessToken
            };
        case USER_LOGIN_ERROR:
            return {
                ...state,
                errorLogin: action.errorLogin
            }
        case USER_LOGOUT_SUCCESS:
            return initState;
        case USER_LOGOUT_ERROR:
            return {
                ...state,
                errorLogout: action.errorLogout
            }
        default:
          return state
    }
};
  
export default authReducer;