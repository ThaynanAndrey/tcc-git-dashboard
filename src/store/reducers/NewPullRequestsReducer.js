import { LOAD_PULL_REQUESTS_NO_PROJECT_SUCCESS, ADDED_PULL_REQUEST_SUCCESS, ADDED_PULL_REQUEST_ERROR } from '../actions/types';

const initState = {
    pullRequestsNoProject: []
};
  
const newPullRequestsReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PULL_REQUESTS_NO_PROJECT_SUCCESS:
            console.log('pull requests no project loaded!');
            return {
                ...state,
                pullRequestsNoProject: action.pullRequestsNoProject
            };
        case ADDED_PULL_REQUEST_SUCCESS:
            console.log("Pull request added");
            return state;
        case ADDED_PULL_REQUEST_ERROR:
            console.log("Error to add pull request");
            return state;
        default:
          return state
    }
};
  
export default newPullRequestsReducer;