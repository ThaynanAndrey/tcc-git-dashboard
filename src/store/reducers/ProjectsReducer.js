import { LOAD_PROJECTS_SUCCESS, LOAD_PROJECTS_ERROR, CREATE_PROJECT_SUCCESS,
    CREATE_PROJECT_ERROR, GET_PROJECT_SUCCESS } from '../actions/types';

export const initState = {
    projects: [],
    project: undefined,
    currentProject: undefined,
    error: undefined
};
  
const projectsReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PROJECTS_SUCCESS:
            return {
                ...state,
                projects: action.projects
            };
        case LOAD_PROJECTS_ERROR:
            return {
                ...state,
                projects: [],
                error: action.error
            };
        case CREATE_PROJECT_SUCCESS:
            return {
                ...state,
                projects: action.projects
            };
        case GET_PROJECT_SUCCESS:
            return {
                ...state,
                project: action.project
            }
        case CREATE_PROJECT_ERROR:
            return state;
        default:
          return state;
    }
};
  
export default projectsReducer;