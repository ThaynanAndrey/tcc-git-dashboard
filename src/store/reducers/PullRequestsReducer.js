import { LOAD_PROJECT_PULL_REQUESTS_SUCCESS, LOAD_PULL_REQUEST_SUCCESS, RESET_PROJECT_PULL_REQUESTS } from '../actions/types';

const initState = {
    projectPullRequests: [],
    selectedPullRequest: { responsavel: {}, repositorio: {} },
};
  
const pullRequestsReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PROJECT_PULL_REQUESTS_SUCCESS:
            console.log('get all pull requests from project!');
            return {
                ...state,
                projectPullRequests: action.projectPullRequests
            };
        case LOAD_PULL_REQUEST_SUCCESS:
            return {
                ...state,
                selectedPullRequest: action.selectedPullRequest
            }
        case RESET_PROJECT_PULL_REQUESTS:
            return {
                ...state,
                projectPullRequests: []
            }
        default:
          return state
    }
};
  
export default pullRequestsReducer;