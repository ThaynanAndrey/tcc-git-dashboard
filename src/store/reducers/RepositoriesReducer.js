import { LOAD_PROJECT_REPOSITORIES_SUCCESS, LOAD_REPOSITORIES_NO_PROJECT_SUCCESS, ADDED_REPOSITORY_SUCCESS,
    ADDED_REPOSITORY_ERROR, LOAD_EXTERNAL_REPOSITORIES, LOAD_EXTERNAL_REPOSITORIES_ERROR, RESET_PROJECT_REPOSITORIES,
    RESET_REPOSITORIES_NO_PROJECT } from '../actions/types';

export const initState = {
    projectRepositories: [],
    repositoriesNoProject: [],
    externalRepositories: [],
    msgExternalRepositories: undefined,
    error: undefined
};
  
const repositoriesReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PROJECT_REPOSITORIES_SUCCESS:
            return {
                ...state,
                projectRepositories: action.projectRepositories
            };
        case LOAD_REPOSITORIES_NO_PROJECT_SUCCESS:
            return {
                ...state,
                repositoriesNoProject: action.repositoriesNoProject
            }
        case ADDED_REPOSITORY_SUCCESS:
            return {
                ...state,
                repositoriesNoProject: action.repositoriesNoProject,
                externalRepositories: action.externalRepositories
            }
        case ADDED_REPOSITORY_ERROR:
            return {
                ...state,
                error: action.error
            }
        case LOAD_EXTERNAL_REPOSITORIES:
        case LOAD_EXTERNAL_REPOSITORIES_ERROR:
            return {
                ...state,
                externalRepositories: action.externalRepositories,
                msgExternalRepositories: action.msgExternalRepositories
            }
        case RESET_PROJECT_REPOSITORIES:
            return {
                ...state,
                projectRepositories: []
            }
        case RESET_REPOSITORIES_NO_PROJECT:
            return {
                ...state,
                repositoriesNoProject: [],
                externalRepositories: []
            }
        default:
          return state
    }
};
  
export default repositoriesReducer;