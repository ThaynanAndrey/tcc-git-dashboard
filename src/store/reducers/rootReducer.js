import { combineReducers } from 'redux';
import authReducer from './AuthReducer';
import pullRequestsReducer from './PullRequestsReducer';
import newPullRequestsReducer from './NewPullRequestsReducer';
import repositoriesReducer from './RepositoriesReducer';

const rootReducer = combineReducers({
    auth: authReducer,
    pullRequests: pullRequestsReducer,
    newPullRequests: newPullRequestsReducer,
    repositories: repositoriesReducer
});

export default rootReducer;