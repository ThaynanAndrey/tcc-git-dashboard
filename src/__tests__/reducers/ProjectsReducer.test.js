import projectsReducer from '../../store/reducers/ProjectsReducer';
import { initState } from '../../store/reducers/ProjectsReducer';

describe('ProjectsReducer', () => {
    it('should return the initial state', () => {
        const state = projectsReducer(undefined, {});
        
        expect(state).toMatchSnapshot();
        expect(state).toEqual(initState);
    });

    it('should handle LOAD_PROJECTS_SUCCESS', () => {
        const projects = [ {name: "Project test", creationDate:"12/12/2012" } ];
        
        const action = {
            type: 'LOAD_PROJECTS_SUCCESS',
            projects
        };
        
        const newState = {
            ...initState,
            projects
        };

        const state = projectsReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle LOAD_PROJECTS_ERROR', () => {        
        const action = {
            type: 'LOAD_PROJECTS_ERROR',
            error: 'Crazy error!'
        };
        
        const newState = {
            ...initState,
            projects: [],
            error: 'Crazy error!'
        };

        const state = projectsReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle CREATE_PROJECT_SUCCESS', () => {
        const projects = [ {name: "New Project test!!!", creationDate:"13/11/2013" } ];
        
        const action = {
            type: 'CREATE_PROJECT_SUCCESS',
            projects
        };
        
        const newState = {
            ...initState,
            projects
        };

        const state = projectsReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle GET_PROJECT_SUCCESS', () => {
        const project = { name: "Lonely test", creationDate:"10/02/2012" };
        
        const action = {
            type: 'GET_PROJECT_SUCCESS',
            project
        };
        
        const newState = {
            ...initState,
            project
        };

        const state = projectsReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle CREATE_PROJECT_ERROR', () => {
        const action = {
            type: 'CREATE_PROJECT_ERROR'
        };

        const state = projectsReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(initState);
    });
});