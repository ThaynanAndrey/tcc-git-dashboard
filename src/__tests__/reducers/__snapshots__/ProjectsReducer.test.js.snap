// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsReducer should handle CREATE_PROJECT_ERROR 1`] = `
Object {
  "currentProject": undefined,
  "error": undefined,
  "project": undefined,
  "projects": Array [],
}
`;

exports[`ProjectsReducer should handle CREATE_PROJECT_SUCCESS 1`] = `
Object {
  "currentProject": undefined,
  "error": undefined,
  "project": undefined,
  "projects": Array [
    Object {
      "creationDate": "13/11/2013",
      "name": "New Project test!!!",
    },
  ],
}
`;

exports[`ProjectsReducer should handle GET_PROJECT_SUCCESS 1`] = `
Object {
  "currentProject": undefined,
  "error": undefined,
  "project": Object {
    "creationDate": "10/02/2012",
    "name": "Lonely test",
  },
  "projects": Array [],
}
`;

exports[`ProjectsReducer should handle LOAD_PROJECTS_ERROR 1`] = `
Object {
  "currentProject": undefined,
  "error": "Crazy error!",
  "project": undefined,
  "projects": Array [],
}
`;

exports[`ProjectsReducer should handle LOAD_PROJECTS_SUCCESS 1`] = `
Object {
  "currentProject": undefined,
  "error": undefined,
  "project": undefined,
  "projects": Array [
    Object {
      "creationDate": "12/12/2012",
      "name": "Project test",
    },
  ],
}
`;

exports[`ProjectsReducer should return the initial state 1`] = `
Object {
  "currentProject": undefined,
  "error": undefined,
  "project": undefined,
  "projects": Array [],
}
`;
